src/lib.rs

```diff
@@ -484,24 +571,30 @@ impl Mastodon {
 
     /// Get statuses of a single account by id. Optionally only with pictures
     /// and or excluding replies.
-    pub fn statuses(&self, id: &str, only_media: bool, exclude_replies: bool)
-        -> Result<Page<Status>>
+    ///
+    /// # Example
+    ///
+    /// ```ignore
+    /// let client = Mastodon::from_data(data);
+    /// let statuses = client.statuses("user-id", None)?;
+    /// ```
+    ///
+    /// ```ignore
+    /// let client = Mastodon::from_data(data);
+    /// let request = StatusesRequest::default()
+    ///                               .only_media();
+    /// let statuses = client.statuses("user-id", request)?;
+    /// ```
+    pub fn statuses<'a, S>(&self, id: &str, request: S) -> Result<Page<Status>>
+            where S: Into<Option<StatusesRequest<'a>>>
     {
-        let mut url = format!("{}/api/v1/accounts/{}/statuses", self.base, id);
+        let url = format!("{}/api/v1/accounts/{}/statuses", self.base, id);
 
-        if only_media {
-            url += "?only_media=1";
-        }
-
-        if exclude_replies {
-            url += if only_media {
-                "&"
-            } else {
-                "?"
-            };
-
-            url += "exclude_replies=1";
-        }
+        let url = if let Some(request) = request.into() {
+            request.to_querystring()
+        } else {
+            url
```

Comment:
The first definition of `url` is no longer needed now correct? So this line should probably be `format!("{}/api/v1/accounts/{}/statuses", self.base, id)`